boa/src/builtins/math/mod.rs

```diff
@@ -415,14 +508,15 @@ impl Math {
     /// [spec]: https://tc39.es/ecma262/#sec-math.imul
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/imul
     pub(crate) fn imul(_: &JsValue, args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
-        Ok(match (
-            args.get(0).map(|x| x.to_u32(context)).transpose()?,
-            args.get(1).map(|x| x.to_u32(context)).transpose()?,
-        ) {
-            (Some(x), Some(y)) => x.wrapping_mul(y) as i32,
-            (_, _) => 0,
-        }
-        .into())
+        // 1. Let a be ℝ(? ToUint32(x)).
+        let x = args.get_or_undefined(0).to_u32(context)?;
+
+        // 2. Let b be ℝ(? ToUint32(y)).
+        let y = args.get_or_undefined(1).to_u32(context)?;
+
+        // 3. Let product be (a × b) modulo 232.
+        // 4. If product ≥ 2^31, return 𝔽(product - 2^32); otherwise return 𝔽(product).
```

Comment:
And here, should it not be `2^32` instead of `2^31`?
```suggestion
        // 4. If product ≥ 2^32, return 𝔽(product - 2^32); otherwise return 𝔽(product).
```

Reply:
Nope, this is to account for negative numbers, since the operation is a multiplication of signed integers